core/src/main/java/cz/xtf/core/openshift/OpenShifts.java

```diff
@@ -169,11 +170,35 @@ private static String getHomeDir() {
 
 	private static String downloadOpenShiftBinary(String version) {
 		String systemType = SystemUtils.IS_OS_MAC ? "macosx" : "linux";
-		String clientLocation = String.format(CLIENTS_URL + "%s/%s/", version, systemType);
+		String clientLocation = null;
+		String ocFileName = "oc.tar.gz";
+
+		if (version.startsWith("3")) {
+			clientLocation = String.format(OCP3_CLIENTS_URL + "%s/%s/", version, systemType);
+		} else {
+			if (StringUtils.isNotEmpty(OpenShiftConfig.version())) {
+				clientLocation = String.format(OCP4_CLIENTS_URL + "%s/%s/", version, systemType);
+			} else {
+				if (SystemUtils.IS_OS_MAC) {
+					systemType = "mac";
+					ocFileName = "oc.zip";
+				} else {
+					systemType = "linux";
+					ocFileName = "oc.tar";
+				}
+				final Optional<ModelNode> downloadsNodeOptional = apiResources("route.openshift.io/v1/routes").get("items").asList().stream().filter(m -> "downloads".equals(m.get("metadata").get("name").asString())).findAny();
```

Comment:
Can `Openshift.getRoutes()` be used here?

Reply:
@mchoma Yes, thx for the tip